apps/src/templates/rubrics/AssessmentsAnnouncementDialog.jsx

```diff
@@ -0,0 +1,85 @@
+import React, {useEffect} from 'react';
+
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import {Heading2, BodyTwoText} from '@cdo/apps/componentLibrary/typography';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import AccessibleDialog from '@cdo/apps/sharedComponents/AccessibleDialog';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import {navigateToHref} from '@cdo/apps/utils';
+import i18n from '@cdo/locale';
+
+import announcementImage from './images/ta-assessments-launch-graphic.jpg';
+
+import styles from './rubrics.module.scss';
+
+export default function AssessmentsAnnouncementDialog() {
+  const [dialogOpen, setDialogOpen] = React.useState(true);
+
+  useEffect(() => {
+    if (dialogOpen) {
+      analyticsReporter.sendEvent(EVENTS.TA_RUBRIC_ANNOUNCEMENT_VIEWED);
+    }
+  });
+
+  if (!dialogOpen) {
+    return null;
+  }
+
+  // post to the server indicating the announcement has been seen.
+  // @return {Promise} A promise that resolves when the post request completes.
+  const postAnnouncementSeen = () => {
+    const url = '/api/v1/users/has_seen_ai_assessments_announcement';
+    return HttpClient.post(url, null, true);
+  };
+
+  const handleClose = () => {
+    // dialog should close immediately, before post request completes
+    setDialogOpen(false);
+    analyticsReporter.sendEvent(EVENTS.TA_RUBRIC_ANNOUNCEMENT_DISMISSED);
+    postAnnouncementSeen();
+  };
+  const handleDismiss = handleClose;
+
+  const handleButtonClick = () => {
+    // wait for the post request to complete before navigating, otherwise the
+    // post request may be cancelled when navigation occurs.
+    analyticsReporter.sendEvent(EVENTS.TA_RUBRIC_ANNOUNCEMENT_CLICKED);
+    postAnnouncementSeen().then(() => {
```

Comment:
Anything we could/should do to gracefully handle a post failure here?

Reply:
I'm not sure what incredibly responsible thing we should do here. the way it is, nothing will happen and we hope the user clicks the button again. perhaps it would be better to take them to the desired page no matter what, and (worst case) the dialog pops open again the next time they return to the unit overivew page? actually, that second option does sound better to me.